lib/events.js

```diff
@@ -425,9 +425,9 @@ EventEmitter.prototype.listeners = function listeners(type) {
     if (!evlistener)
       ret = [];
     else if (typeof evlistener === 'function')
-      ret = [evlistener];
+      ret = [evlistener.listener || evlistener];
     else
-      ret = arrayClone(evlistener, evlistener.length);
```

Comment:
Should we remove the `arrayClone` function and do replace all `arrayClone()` to `unwrapListeners`?